carapace-server/src/main/java/org/carapaceproxy/server/config/RouteConfiguration.java

```diff
@@ -55,11 +56,11 @@ public RequestMatcher getMatcher() {
         return matcher;
     }
 
-    public boolean matches(HttpRequest request) {
+    public boolean matches(RequestHandler handler) throws MatchingException {
         if (!enabled) {
             return false;
         }
-        return matcher.matches(request);
+        return matcher.matches(new RequestMatchingContext(handler));
```

Comment:
if you make RequestHandler implement RequestMatchingContext we will save an allocation an some pointer accesses